clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java

```diff
@@ -618,9 +621,7 @@ public boolean hasAnySubscription() {
   public synchronized void resetPartitionConsumptionOffset(PubSubTopicPartition topicPartition) {
     throwIfNotRunning();
     amplificationFactorAdapter.execute(topicPartition.getPartitionNumber(), subPartition -> {
-      partitionToPendingConsumerActionCountMap.computeIfAbsent(subPartition, x -> new AtomicInteger(0))
-          .incrementAndGet();
-      consumerActionsQueue.add(
+      processConsumerActionOrEnqueue(
           new ConsumerAction(
               RESET_OFFSET,
```

Comment:
In old implementation, there seems to have some special handling of `RESET_OFFSET` in StoreIngestionTask.internalClose(), in case that StoreIngestionTask is killed, but `RESET_OFFSET` is in queue but not yet processed. Shall we also handle this case?